### ISSUE TITLE
Allow running custom commands on selection/groups; akin to `find -exec`

### ISSUE THREAD

**Issue by zoickx** [qarmin/czkawka#827]
In addition to the standard `Move`, `Delete`, etc., actions, it would be nice to have an option to run custom commands on the files (or groups of files) selected by Czkawka.[^1]

#### Motivation

While probably a feature targeted at more advanced uses and users, I feel it would be a really powerful addition at very little cost. _A lot_ of the tasks currently unsolvable in Czkawka (open issues like #162, #751, even #241, and probably much more[^2]) could be solved by a cleverly-applied bash one-liner.

What if the user wants more actions than "move", "link", "delete"? Any number of desirable actions are possible for duplicate, old or otherwise broken files. For example, one might want to update file metadata, add watermarks to duplicates, add old files to a compressed archive, merge files, back them up to some cloud, etc.

Currently Czkawka excels at _finding what needs action_, but the _actions themselves_ feel very limited. Allowing the user to run their own scripts on Czkawka's output would alleviate this issue entirely.

And while yes, it is already possible to parse Czkawka's output files and work from there, it's not nearly as convenient.

#### Suggestion

I see two approaches to this, the second one being more powerful, but possibly harder to implement.

1. **Pass one filename**
Just™ add a button `Run command` next to `Move`, prompt the user for a piece of shell script, and run the provided command on each selected file individually. To me this sounds like 80% of the benefits for 20% of the work. Whatever is currently used to move a file, could just as well be used to execute a shell with a given command on that file.

    In this case, for example, #751, would be solved by running a script `fix-extension.sh` on each found file.

    This option is limited, though, because
a) The right selection must be done first from Czkawka - no custom logic (so doesn't solve #241)
b) The action is context-independent - no complex actions based on other info about the file's group

2. **Pass the full group**[^3] (where applicable)
Instead of selecting files from groups in Czkawka and executing actions on each one individually, it would be nice to be able to only group files in Czkawka and leave both selection and action to the script.

    This would allow arbitrary logic to be applied to decide what to do with the group: for example, #241 could be resolved this way. Having a list of files (say a list of duplicate images to select from), one can stat them for filesize, resolution, timestamps, exif data, etc., and prioritize accordingly in the script itself - no need to select or pass that info from Czkawka.

[^1]: In a sense, this would be similar to the really powerful [`find -exec`](https://linuxhint.com/find-exec-command-linux/), with Czkawka itself being the `find`, except instead of dumb searches for names, we search for duplicates and other junk.

[^2]: To be clear, I'm not advocating for shell commands to replace other methods - this is not literally *closing* issues, but just demonstrating the power of the approach on some existing examples.

[^3]: Other info could be useful if possible: whether files are marked original, whether they were already selected in Czkawka, maybe some way of passing similarity levels where applicable, though this seems rather complicated.
